src/WebClient.spec.js

```diff
@@ -201,18 +201,28 @@ describe('WebClient', function () {
 
         // intentially vague about the method name
         return this.client.apiCall('upload', {
-            file: imageBuffer,
-            filename: 'train.png',
-          })
+          file: imageBuffer,
+          filename: 'train.png',
+        })
+          .then((parts) => {
+            assert.lengthOf(parts.files, 1);
+            const file = parts.files[0];
+            // options were not provided to the form builder
+            assert.include(file, { fieldname: 'file' });
```

Comment:
still not asserting that the `filename` field exists, and removed the comment that would remind us to improve the tests so that it is asserted. why?

Reply:
sorry, I've overseen this assertion statement and felt that the fix solved the problem pointed in the comment. I'll bring your TODO and assertion back.